game-core/src/main/java/games/strategy/debug/ErrorMessage.java

```diff
@@ -26,6 +26,7 @@
  * to be user friendly, clicking 'show details' would show full details of all error messages.
  * </p>
  */
+@SuppressWarnings("ImmutableEnumChecker") // Enum singleton pattern
```

Comment:
The comment is notable. I recall instances of similar in other projects where the explanation was a bit more verbose, but we had similar with the same explanation being copy/pasted. It turned out to be pretty boiler plate in the end and added to the overhead of having the suppress warning annotation. In this case it's okay, but the pattern of commenting on suppress warnings with after line comments I think could become inconsistent/harmful.

What do you think of us creating a doc page with a registry of warning suppression? There we could give more details about what we are suppressing and why. Then if there are any oddball cases, the comment next to the warning suppression would be more noteworthy rather than boiler plate (which maybe is not super helpful if you are not familar with the enum singleton pattern, or where/why we are suppressing the warning).

Reply:
> the pattern of commenting on suppress warnings with after line comments I think could become inconsistent/harmful

Ideally, `@SuppressWarnings` would include a `reason` parameter so the comment would be available to tooling.  But it doesn't. :disappointed: 

> What do you think of us creating a doc page with a registry of warning suppression?

Pro: As you said, it normalizes everything to avoid copy/paste.
Con: It will probably get quickly out-of-date.

At this point, we have less than 10 warning suppressions other than "unused" or "unchecked", so a registry may be overkill.  Maybe we should define a threshold and revisit this topic in the future if we ever exceed it.

Reply:
> Ideally, @SuppressWarnings would include a reason parameter so the comment would be available to tooling. But it doesn't.

Good point, but if the comment is the same each time the suppression is applied, then it is more intrinsic to the suppression itself.  

I'm mostly concerned that this will be a pattern for the next iteration, and maybe it'll be hard to work it in; which encourages comment clutter. 

Not necessarily that big of a concern though. I suppose only worth briefly considering if the comments are really helpful. The fact they are for error prone would be the most helpful for me.